src/dmd/mars.d

```diff
@@ -574,7 +574,15 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
     }
 
     if (params.addMain && !global.hasMainFunction)
-        modules.push(moduleWithEmptyMain());
+    {
+        auto mainModule = moduleWithEmptyMain();
+        modules.push(mainModule);
+
+        if (!params.oneobj)
+            params.objfiles.push(mainModule.objfile.toChars());
+        else if (modules.length == 1)
+            params.objfiles.insert(0, mainModule.objfile.toChars());
```

Comment:
Inserting at position 0 is just an LDC specificum for `-singleobj`, DMD shouldn't need it and probably just append it, as for the `!oneobj` case.

Reply:
There are some `.debug_ranges` conflicts on debug info for `-main` generated functions. Perhaps, is this somewhat related?